code/go/0chain.net/blobbercore/handler/health.go

```diff
@@ -1,29 +1,44 @@
 package handler
 
 import (
+	"sync"
+
 	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/config"
 	"github.com/0chain/blobber/code/go/0chain.net/core/logging"
 	"github.com/0chain/blobber/code/go/0chain.net/core/transaction"
 	"go.uber.org/zap"
 )
 
-var blobberHealthCheckErr error
+var (
+	// blobberHealthCheckError use it on stats page
+	blobberHealthCheckError error
+	blobberHealthCheckMutex sync.RWMutex
+)
+
+func setBlobberHealthCheckError(err error) {
+	blobberHealthCheckMutex.Lock()
+	defer blobberHealthCheckMutex.Unlock()
 
-func SetBlobberHealthError(err error) {
-	blobberHealthCheckErr = err
+	blobberHealthCheckError = err
 }
 
-func GetBlobberHealthError() error {
-	return blobberHealthCheckErr
+func getBlobberHealthCheckError() error {
+	blobberHealthCheckMutex.Lock()
+	defer blobberHealthCheckMutex.Unlock()
```

Comment:
Do not use `defer` here

Reply:
inline defer performance seems be fixed on latest go version.

https://rakyll.org/inlined-defers/

Reply:
@peterlimg   how to lock for reading is better if don't use `defer`?. below solution is ok?
```
func getBlobberHealthCheckError() error {
	blobberHealthCheckMutex.RLock()
        err :=  blobberHealthCheckError
        blobberHealthCheckMutex.RUnlock()
	return err
}

```

Reply:
fixed